lib/decidim/git_backport_manager.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "English"
+
+module Decidim
+  class GitBackportManager
+    def initialize(pull_request_id:, release_branch:, backport_branch:, working_dir: Dir.pwd)
+      @pull_request_id = pull_request_id
+      @release_branch = release_branch
+      @backport_branch = backport_branch
+      @working_dir = working_dir
+    end
+
+    def call
+      Dir.chdir(working_dir) do
+        exit_if_unstaged_changes
```

Comment:
I personally did not enjoy this check as many times I have some unrelated unstaged changes that I wanna keep regardless of the backport.

You are able to do the cherry-pick also when you have unstaged changes but I can also see why you did this. Changing the branch might fail sometimes in case we have conflicting unstaged changes which can cause issues with the backports.

So I understand why you want to do this in an automated script, I just wanted to point out here I did not enjoy it personally.

Reply:
This was a pain for testing the script for sure, I had to comment this line out to try it.

I have to say that in the normal usage, I didn't feel the pain. I use (and abuse) `git stash`, but I understand that this at least should be an option from the user. So, I'm adding the possibility to skip this behavior with a flag (`[--exit-with-unstaged-changes], [--no-exit-with-unstaged-changes]`)